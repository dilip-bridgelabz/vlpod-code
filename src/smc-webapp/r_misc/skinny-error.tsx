/*
 *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
 *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
 */

import * as React from "react";
import { SimpleX } from "./simple-x";

interface Props {
  error_text: string;
  on_close: () => void;
}

export function SkinnyError({ error_text, on_close }: Props) {
  return (
    <div style={{ color: "red" }}>
      <SimpleX onClick={on_close} /> {error_text}
    </div>
  );
}
