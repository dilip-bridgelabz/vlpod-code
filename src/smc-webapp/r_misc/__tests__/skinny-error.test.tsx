/*
 *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
 *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
 */

import * as React from "react";
import { shallow } from "enzyme";
import { SkinnyError } from "../skinny-error";

test("smoke test", () => {
  const rendered = shallow(
    <SkinnyError on_close={() => undefined} error_text={"Error testing text"} />
  );
  expect(rendered).toMatchSnapshot();
});

test("test interaction", () => {
  const mock_on_close = jest.fn();
  const rendered = shallow(
    <SkinnyError on_close={mock_on_close} error_text={"Error testing text"} />
  );

  const evt = new MouseEvent("click");

  rendered.children().first().simulate("click", evt);
  expect(mock_on_close.mock.calls.length).toBe(1);
});
