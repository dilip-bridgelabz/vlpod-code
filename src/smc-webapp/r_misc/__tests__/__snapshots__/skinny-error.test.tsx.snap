// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke test 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  <SimpleX
    onClick={[Function]}
  />
   
  Error testing text
</div>
`;
